### PR TITLE
AV-127273 add document for AMKO multitenancy

### PR DESCRIPTION
**Pull request by arihantg**

### CODE REVIEW

File: docs/roles/ako-tenant.json [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        {
+            "type": "READ_ACCESS",
+            "resource": "PERMISSION_GSLB"
+        },
+        {
+            "type": "WRITE_ACCESS",
+            "resource": "PERMISSION_GSLBSERVICE"
+        },
+        {
+            "type": "WRITE_ACCESS",
```
Comment (swathinsankaran) [inline docs/roles/ako-tenant.json:242]: I think we have to give `READ_ACCESS` instead of `WRITE_ACCESS` here.

File: docs/amko_tenancy.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Tenancy support in AMKO
+
+This feature allows AMKO to create GSLB object in a user-specific tenant in Avi.The expected isolation and administrative restrictions of a multi-tenant architecture in NSX Advanced Load Balancer GSLB extend to AMKO
```
Comment (apalsule) [inline docs/amko_tenancy.md:3]: user-specific -> user specified

File: docs/amko_tenancy.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+## Tenant Context
+
+AVI non admin tenants primarily operate in 2 modes, **provider context** and **tenant context**.
+
+### Provider Context
```
Comment (apalsule) [inline docs/amko_tenancy.md:9]: I dont think this needs to be mentioned here, I dont see an impact on AMKIO for this

File: docs/amko_tenancy.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+### Tenant Context
+
+Service Engines are isolated from `admin` tenant. A new `Default-Group` is created within the tenant. All the objects including Service Engines are created in tenant context. Requires `config_settings.se_in_provider_context` flag to be set to `False` when creating tenant. 
+
+## Steps to enable Tenancy in AMKO
+
+In this example we will run AMKO in `billing` tenant.
+
+### 1. Install AKO in required Tenant.
```
Comment (apalsule) [inline docs/amko_tenancy.md:21]: Its not clear from teh documentation whether AKO and AMKO deployments need to be in the same tenant or can be in different ones

File: docs/crds/gslbconfig.md [hunk 5]
```diff
@@ -26,10 +27,11 @@ spec:
...
 6. `gslbLeader.controllerVersion`: The version of the GSLB leader cluster.
 7. `gslbLeader.controllerIP`: The GSLB leader IP address or the hostname along with the port number, if any.
-8. `memberClusters`: The kubernetes/openshift cluster contexts which are part of this GSLB cluster. See [here](../kubeconfig.md#creating-a-multi-cluster-kubeconfig-file) to create contexts for multiple kubernetes clusters.
-9.  `refreshInterval`: This is an internal cache refresh time interval, on which syncs up with the AVI objects and checks if a sync is required.
-10. `logLevel`: Define the log level that the amko pod prints. The allowed levels are: `[INFO, DEBUG, WARN, ERROR]`.
-11. `useCustomGlobalFqdn`: If set to true, AMKO will look for AKO HostRules to derive the GslbService name using the local to global fqdn mapping. If set to false (default case), AMKO ignores AKO HostRules and uses the default way of deriving GslbService names by just looking at the local fqdn in the ingress/route/service type LB. See [Local and Global Fqdn](../local_and_global_fqdn.md).
+8. `gslbLeader.tenant`: The tenant where all the AKO objects will be created in AVI.
```
Comment (apalsule) [inline docs/crds/gslbconfig.md:30]: Is this AKO or AMKO objects ? If AMKO, say AMKO will create all GS's in this tenant.

File: docs/roles/ako-tenant.json [hunk 6]
```diff
@@ -0,0 +1,250 @@
+{
```
Comment (apalsule) [inline docs/roles/ako-tenant.json:1]: Why is this ako-tenant.json, shouldn't it be amko-tenant ?

File: docs/roles/ako-tenant.json (hunk 1 above)
Comment (arihantg) [reply docs/roles/ako-tenant.json]: updated with READ_ACCESS

File: docs/amko_tenancy.md (hunk 2 above)
Comment (arihantg) [reply docs/amko_tenancy.md]: changed

File: docs/amko_tenancy.md (hunk 3 above)
Comment (arihantg) [reply docs/amko_tenancy.md]: removed this section

File: docs/crds/gslbconfig.md (hunk 5 above)
Comment (arihantg) [reply docs/crds/gslbconfig.md]: updated

File: docs/roles/ako-tenant.json (hunk 6 above)
Comment (arihantg) [reply docs/roles/ako-tenant.json]: it should be amko-tenant. Modified accordingly.

File: docs/amko_tenancy.md (hunk 4 above)
Comment (arihantg) [reply docs/amko_tenancy.md:8]: mentioned explicityly AKO and AMKO needs to be in same tenant

**Review by apalsule — APPROVED** [review]

**Pull request merged**